src/module/rolls/preroll-dialog.ts

```diff
@@ -0,0 +1,320 @@
+  }
+
+  static async showForOfficialMove(moveDfId: string, actor: IronswornActor) {
+    const moveItem = await getFoundryMoveByDfId(moveDfId)
+    if (!moveItem) {
+      throw new Error(`Couldn't find item for move '${moveDfId}'`)
+    }
+
+    return this.showForMoveItem(moveItem, actor, { moveDfId })
+  }
+
+  static async showForCustomMove(move: IronswornItem, actor: IronswornActor) {
+    if (move.type !== 'sfmove') {
+      throw new Error('this only works with SF moves')
+    }
+
+    return this.showForMoveItem(move, actor, { moveId: move.id || undefined })
+  }
+
+  private static async showForMoveItem(
+    move: IronswornItem,
+    actor: IronswornActor,
+    prerollOptions: PreRollOptions
+  ) {
+    const data = move.data as SFMoveDataProperties
+    const options = rollableOptions(data.data.Trigger)
+    if (!options.length) {
+      throw new Error(
+        `Move '${move.name}' (${JSON.stringify(
+          prerollOptions
+        )}) is not rollable`
+      )
+    }
+
+    const title = move.name || 'MOVE'
+    prerollOptions.momentum = (
+      actor.data as CharacterDataProperties
+    ).data.momentum
+
+    const content = await this.renderContent({
+      prerollOptions,
+      move,
+      actor,
+      action: true,
+    })
+    const buttons = {}
+    const addButton = (i: number, mode: RollMethod, stats: string[]) => {
+      // TODO: i18n
+      const label = mode === 'Any' ? stats[0] : `${mode} of ${stats.join(', ')}`
+
+      buttons[i.toString()] = {
+        // use the below instead as a silly method for sneaking classes in
+        // buttons[
+        //   kebabCase(label) + ' clickable isicon-d10-tilt juicy icon-button'
+        // ] = {
+        label,
+        icon: '<i class="isicon-d10-tilt juicy"></i>',
+        callback: (el: HTMLElement | JQuery<HTMLElement>) => {
+          prerollOptions.action = resolvedStatForMode(mode, stats, actor)
+          console.log(prerollOptions)
+          IronswornPrerollDialog.submitRoll(el, prerollOptions)
+        },
+      }
+    }
+
+    for (let i = 0; i < options.length; i++) {
+      const option = options[i]
+      addButton(i, option.Method, option.Using)
+    }
+
+    return new IronswornPrerollDialog(prerollOptions, {
+      title,
+      content,
+      buttons,
+      default: '0',
+    }).render(true)
+  }
+
+  private static submitRoll(
+    el: HTMLElement | JQuery<HTMLElement>,
+    opts: PreRollOptions
+  ) {
+    const form = el[0].querySelector('form')
+    console.log(form, opts)
+
+    // Manual adds; only for action rolls
+    if (opts.action) {
+      opts.adds = parseInt(form.adds.value || '0', 10)
+    }
+
+    if (form.automaticOutcome.checked) {
+      opts.automaticOutcome = {
+        source: 'set manually',
+        value: form.automaticOutcomeValue.value,
+      }
+    }
+    if (form.presetActionDie.checked) {
+      opts.presetActionDie = {
+        source: 'set manually',
+        value: parseInt(form.presetActionDieValue.value || '0'),
```

Comment:
hmm. wouldn't `form.presetActionDieValue.valueAsNumber` be simpler? or is this just a workaround for some weirdness with foundry?

Reply:
Oh TIL this exists 😄